src/cpu/aarch64/acl_indirect_gemm_convolution.hpp

```diff
@@ -0,0 +1,176 @@
+
+struct acl_indirect_gemm_resource_t : public resource_t {
+
+    acl_indirect_gemm_resource_t()
+        : acl_obj_(utils::make_unique<acl_obj_t<arm_compute::NEGEMMConv2d>>()) {
+    }
+
+    status_t configure(const acl_conv_conf_t &acp) {
+        if (!acl_obj_) return status::out_of_memory;
+
+        // Init Compute Library tensors based on info from descriptor
+        acl_obj_->src_tensor.allocator()->init(acp.src_info);
+        acl_obj_->wei_tensor.allocator()->init(acp.wei_info);
+        acl_obj_->dst_tensor.allocator()->init(acp.dst_info);
+        acl_obj_->bia_tensor.allocator()->init(acp.bia_info);
+
+        // clang-format off
+        acl_obj_->conv.configure(
+            &acl_obj_->src_tensor,
+            &acl_obj_->wei_tensor,
+            acp.with_bias ? &acl_obj_->bia_tensor : nullptr,
+            &acl_obj_->dst_tensor,
+            arm_compute::Conv2dInfo(acp.padstride_info,
+                                    acp.dilation_info,
+                                    acp.act_info,
+                                    false,
+                                    1));
+        // clang-format on
+
+        return status::success;
+    }
+
+    acl_obj_t<arm_compute::NEGEMMConv2d> &get_acl_obj() const {
+        return *acl_obj_;
+    }
+
+    DNNL_DISALLOW_COPY_AND_ASSIGN(acl_indirect_gemm_resource_t);
+
+private:
+    std::unique_ptr<acl_obj_t<arm_compute::NEGEMMConv2d>> acl_obj_;
+
+}; // acl_indirect_gemm_resource_t
+
+struct acl_indirect_gemm_convolution_fwd_t : public primitive_t {
+
+    struct pd_t : public cpu_convolution_fwd_pd_t {
+        pd_t(const convolution_desc_t *adesc, const primitive_attr_t *attr,
+                const typename pd_t::base_class *hint_fwd_pd)
+            : cpu_convolution_fwd_pd_t(adesc, attr, hint_fwd_pd), acp_() {}
+
+        DECLARE_COMMON_PD_T("indirect_gemm:acl",
+                acl_indirect_gemm_convolution_fwd_t, USE_GLOBAL_SCRATCHPAD);
+
+        status_t init(engine_t *engine) {
+            using namespace data_type;
+            using smask_t = primitive_attr_t::skip_mask_t;
+
+            bool ok = is_fwd()
+                    && set_default_alg_kind(alg_kind::convolution_direct)
+                    && expect_data_types(data_type::f32, data_type::f32,
+                            data_type::f32, data_type::f32, undef)
+                    && !has_zero_dim_memory()
+                    && attr()->has_default_values(
+                            smask_t::post_ops, data_type::f32)
+                    && post_ops_ok();
+            if (!ok) return status::unimplemented;
+
+            auto conf_status = acl_convolution_utils::init_conf_indirect_gemm(
+                    acp_, src_md_, weights_md_, dst_md_, bias_md_, *desc(),
+                    *attr());
+            if (conf_status != status::success) return status::unimplemented;
+
+            // Number of threads in Compute Library is set by OMP_NUM_THREADS
+            // dnnl_get_max_threads() == OMP_NUM_THREADS
```

Comment:
Just a side note, please be aware of oneDNN Theadpool threading, in that case amount of threads at execution can be different from amount of threads at creation time: https://oneapi-src.github.io/oneDNN/dev_guide_threadpool.html

Reply:
Many thanks for this remark, we have been considering only the case DNNL_CPU_RUNTIME=OMP so far.